File: LeetCode/Math/Python/Leetcode396-旋转函数.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        acc,n,res = sum(nums),len(nums),0
        for i in range(n):
            res += i * nums[i]
        f = res 
        for i in range(n - 1,-1,-1):
            f += acc - n * nums[i]
            res = max(res,f)
        return res

File: LeetCode/Math/Rust/Leetcode396-旋转函数.rs
impl Solution {
  pub fn max_rotate_function(nums: Vec<i32>) -> i32 {
    let acc: i32 = nums.iter().sum();
    let n = nums.len();
    let mut res = 0;
    for i in 0..n {
      res += nums[i] * (i as i32);
    }
    let mut f = res;
    for i in (0..n).rev() {
      f += acc - (n as i32) * nums[i];
      res = res.max(f);
    }
    res
  }
}
